src/sympc/tensor/mpc_tensor.py

```diff
@@ -663,10 +663,26 @@ def __apply_private_op(
             )
 
         if op_str in {"mul", "matmul", "conv2d", "conv_transpose2d"}:
+            from sympc.protocol import Falcon
             from sympc.protocol.spdz import spdz
+            from sympc.tensor import ReplicatedSharedTensor
+
+            if self.session.protocol.share_class == ShareTensor:
+
+                result = spdz.mul_master(self, y, op_str, kwargs_)
+                result.shape = MPCTensor._get_shape(op_str, self.shape, y.shape)
+
+            elif self.session.protocol.share_class == ReplicatedSharedTensor:
+
+                result = Falcon.mul(self, y, self.session)
+                result = MPCTensor(
+                    shares=result, shape=self.shape, session=self.session
```

Comment:
MAYBE : for shape you could use `MPCTensor._get_shape` as it changes during the operation.

Reply:
result._get_shape?

Reply:
Something like this:
https://github.com/OpenMined/SyMPC/blob/2005f3393c03296bbdf18cecafc9c0582783e84f/src/sympc/tensor/mpc_tensor.py#L769

Reply:
Alright then. Good catch.